src/scripts/browserDeviceProfile.js

```diff
@@ -220,11 +220,17 @@ function supportsHdr10(options) {
     return options.supportsHdr10 ?? (false // eslint-disable-line sonarjs/no-redundant-boolean
             || browser.tizen
             || browser.web0s
-            || browser.safari && ((browser.iOS && browser.iOSVersion >= 11) || browser.osx)
-            // Chrome mobile and Firefox have no client side tone-mapping
+            // All browsers on iOS are required to use WebKit underneath. HDR support in browsers was added in iOS 13.4 when Safari 13.1 was released.
+            //  * https://developer.apple.com/documentation/safari-release-notes/safari-13_1-release_notes
+            || browser.iOS && browser.iOSVersion >= 13.4
+            // Chrome mobile have no client side tone-mapping
             // Edge Chromium 121+ fixed the tone-mapping color issue on Nvidia
             || browser.edgeChromium && (browser.versionMajor >= 121)
-            || browser.chrome && !browser.mobile
+            // Chrome on mobile now has client side tone-mapping. Tested on Chrome Android 128 but first support version is unknown.
+            || browser.chrome && !browser.iOS && browser.mobile && (browser.versionMajor >= 128)
+            // Firefox 100+ has support for HDR on macOS/OS X. It requires OS support, which was added in macOS 10.15 Catalina.
+            //  * https://www.mozilla.org/en-US/firefox/100.0/releasenotes/
+            || browser.firefox && browser.osx && (!browser.iphone && !browser.ipod && !browser.ipad) && (browser.osxVersion >= 10.15) && (browser.versionMajor >= 100)
```

Comment:
Please don't. The support is very bad and even direct playing an mp4 file will result in washed out color as firefox does not do client-side tone mapping and does not trigger HDR colors in all cases. It seems like such support is never completed on platforms other than YouTube.

Reply:
@gnattu I tested this on my macOS 14.6 and Firefox 129.0.1. Is there a particular setup you're running into that's breaking for you?

All files here seem to work for me: https://drive.google.com/drive/folders/1dSPUmJ_Y99-VL6TxfndPbISyUfRJ3SAY

Reply:
> Is there a particular setup you're running into that's breaking for you?

It is having problems identifying HDR files so you are lucky that all your files are working because some of them are not. For example, this sample video from LG will trigger HDR playback in both Safari and Chrome but not in Firefox:

https://mega.nz/file/5lsGUYpT#B_KNF251rODrq5lWwGYHxSuBsiqEZQsGCRCWO83iAdY

Reply:
This is because the video provided above is in an unusual format of VP9 + AAC in an mp4 container. Firefox's parsing logic is stricter about the container <-> codec mappings. This behavior isn't unique to Firefox -- iOS's browsers also have a history of not playing nice with VP9 in mp4 containers either.

I will create a separate change to fix this specifically for Firefox and break up this change into individual pieces.